src/UrlGenerator.php

```diff
@@ -124,6 +111,32 @@ public function generateAbsolute(
         return $uri === null ? $url : $this->generateAbsoluteFromLastMatchedRequest($url, $uri, $scheme);
     }
 
+    public function generateFromCurrent(array $replacedParams, string $fallbackRouteName = null): string
+    {
+        if ($this->currentRoute === null || $this->currentRoute->getName() === null) {
+            if ($fallbackRouteName !== null) {
+                $fallbackRoute = $this->routeCollection->getRoute($fallbackRouteName);
+                return $this->generate($fallbackRouteName, $replacedParams);
+            }
+
+            if ($this->currentRoute->getUri() !== null) {
```

Comment:
Here can be error if `$this->currentRoute` is `null`.

Reply:
Done